apps/nextjs/src/app/page.tsx

```diff
@@ -33,10 +32,21 @@ export default async function HomePage() {
               </div>
             }
           >
-            <PostList posts={posts} />
+            <PostListWrapped />
           </Suspense>
         </div>
       </div>
     </main>
   );
 }
+
+async function PostListWrapped() {
+  const posts = await api.post.all();
+  setQueryData((t) => [t.post.all, undefined], posts); // would be cool if this was automatically put to the cache
+
+  return (
+    <HydrateClient>
```

Comment:
2. If it was `posts` were branded, we could do some helper that automatically knew how to add it in the RQ cache, like

```tsx
<HydrdateClient queries={[ posts ]}>
   <PostList />
</HydrateClient>
```

Reply:
Or maybe it could be something altogether automagic similar to https://tanstack.com/query/latest/docs/framework/react/guides/advanced-ssr#experimental-streaming-without-prefetching-in-nextjs

and we'd just have a root `<Hydrate>` thing in the layout that somehow had a stream of data it pushed to the cache (not sure if it's technically feasible unless we can hack RSC<->client comms to work with streams or async iterators